src/ray/raylet/node_manager.cc

```diff
@@ -957,6 +1017,38 @@ void NodeManager::HandleObjectMissing(const ObjectID &object_id) {
   }
 }
 
+void NodeManager::ForwardTaskOrResubmit(const Task &task,
+                                        const ClientID &node_manager_id) {
+  /// TODO(rkn): Should we check that the node manager is remote and not local?
+  /// TODO(rkn): Should we check if the remote node manager is known to be dead?
+  const TaskID task_id = task.GetTaskSpecification().TaskId();
+
+  // Attempt to forward the task.
+  if (!ForwardTask(task, node_manager_id).ok()) {
+    RAY_LOG(INFO) << "Failed to forward task " << task_id << " to node manager "
+                  << node_manager_id;
+
+    // Create a timer to resubmit the task in a little bit.
+    boost::asio::deadline_timer timer(io_service_);
+    auto retry_duration = boost::posix_time::milliseconds(
+        RayConfig::instance()
+            .node_manager_forward_task_retry_timeout_milliseconds());
+    timer.expires_from_now(retry_duration);
+    timer.async_wait(
+        [this, task, task_id](const boost::system::error_code &error) {
+          // Timer killing will receive the boost::asio::error::operation_aborted,
+          // we only handle the timeout event.
+          if (!error) {
+            RAY_LOG(INFO) << "In ForwardTask retry callback for task " << task_id;
+            // TODO(rkn): What will this do to the lineage cache? It
+            // will re-add the task to the lineage cache, which is
+            // probably not what we want.
+            SubmitTask(task, Lineage());
```

Comment:
@stephanie-wang please take a look at this line/comment.

Reply:
I think we probably want the `EnqueuePlaceableTask` method instead?